beacon-chain/core/epoch/precompute/new.go

```diff
@@ -46,6 +47,9 @@ func New(ctx context.Context, state *stateTrie.BeaconState) ([]*Validator, *Bala
 
 		vp[idx] = p
 		return nil
-	})
+	}); err != nil {
+		log.WithError(err).Error("Failed to initialize precompute")
```

Comment:
shouldn't we instead return an error in `New` ?

Reply:
I was trying to avoid any refactoring, but maybe this one is small. I agree with you and I'll give this a try

Reply:
This is done now, Thanks!